Assets/_Project/Scripts/UserInterface/Popups/PopupManager.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using JetBrains.Annotations;
+using PolSl.UrbanHealthPath.UserInterface.Initializers;
+using PolSl.UrbanHealthPath.UserInterface.Interfaces;
+using UnityEngine;
+
+namespace PolSl.UrbanHealthPath.UserInterface.Popups
+{
+    public class PopupManager : MonoBehaviour
+    {
+        public PopupType CurrentPopupType { get; private set; }
+
+        [Serializable] public struct Popup
+        {
+            [SerializeField] private PopupType type;
+            [SerializeField] private GameObject popupObject;
+
+            public PopupType GetPopupType()
+            {
+                return type;
+            }
+
+            public GameObject GetPopupObject()
+            {
+                return popupObject;
+            }
+        }
+
+        [SerializeField] private Popup[] popupsWithTypes;
+
+        private Dictionary<PopupType, GameObject> _popups;
+
+        private GameObject _currentPopup;
+
+        private static PopupManager _instance;
```

Comment:
No Singleton needed here. We'll use Dependency Injection and pass the instance of this class where it is needed.

Reply:
Changed -> our changes require the initializaiton to add an event to a station view which will close pop up before opening a new one - this change can only be done after merge.